packages/engine/Source/Scene/I3SSublayer.js

```diff
@@ -0,0 +1,177 @@
+import Check from "../Core/Check.js";
+import defaultValue from "../Core/defaultValue.js";
+import defined from "../Core/defined.js";
+import I3SDataProvider from "./I3SDataProvider.js";
+import I3SLayer from "./I3SLayer.js";
+import Resource from "../Core/Resource.js";
+
+/**
+ * This class implements an I3S sublayer for Building Scene Layer.
+ * <p>
+ * This object is normally not instantiated directly, use {@link I3SSublayer.fromData}.
+ * </p>
+ * @alias I3SSublayer
+ * @internalConstructor
+ */
+function I3SSublayer(dataProvider, parent, sublayerData) {
+  this._dataProvider = dataProvider;
+  this._parent = parent;
+  this._data = sublayerData;
+  this._name = sublayerData.name;
+  this._modelName = sublayerData.modelName;
+  this._visibility = defaultValue(sublayerData.visibility, true);
+  this._resource = undefined;
+  this._sublayers = [];
+  this._i3sLayers = [];
+}
+
+Object.defineProperties(I3SSublayer.prototype, {
+  /**
+   * Gets the resource for the sublayer
+   * @memberof I3SSublayer.prototype
+   * @type {Resource}
+   * @readonly
+   */
+  resource: {
+    get: function () {
+      return this._resource;
+    },
+  },
+
+  /**
+   * Gets the I3S data for this object.
+   * @memberof I3SSublayer.prototype
+   * @type {object}
+   * @readonly
+   */
+  data: {
+    get: function () {
+      return this._data;
+    },
+  },
+
+  /**
+   * Gets the name for the sublayer.
+   * @memberof I3SSublayer.prototype
+   * @type {string}
+   * @readonly
+   */
+  name: {
+    get: function () {
+      return this._name;
+    },
+  },
+
+  /**
+   * Gets the model name for the sublayer.
+   * @memberof I3SSublayer.prototype
+   * @type {string}
+   * @readonly
+   */
+  modelName: {
+    get: function () {
+      return this._modelName;
+    },
+  },
+
+  /**
+   * Gets the collection of child sublayers.
+   * @memberof I3SSublayer.prototype
+   * @type {I3SSublayer[]}
+   * @readonly
+   */
+  sublayers: {
+    get: function () {
+      return this._sublayers;
+    },
+  },
+
+  /**
+   * Gets or sets the sublayer visibility.
+   * @memberof I3SSublayer.prototype
+   * @type {boolean}
+   */
+  visibility: {
+    get: function () {
+      return this._visibility;
+    },
+    set: function (value) {
+      //>>includeStart('debug', pragmas.debug);
+      Check.defined("value", value);
+      //>>includeEnd('debug');
+
+      this._visibility = value;
+      for (let i = 0; i < this._i3sLayers.length; i++) {
+        this._i3sLayers[i]._updateVisibility();
+      }
+    },
+  },
+
+  /**
+   * Determines if the sublayer will be shown.
+   * @memberof I3SSublayer.prototype
+   * @type {boolean}
+   * @readonly
+   */
+  show: {
```

Comment:
What's the difference between `visibility` and `show`?

Reply:
Visibility is a state of the sublayer checkbox, while show returns whether the sublayer is visible or not depending on its visibility and visibility of all its parents.
Building scene layers consist of a hierarchy of sublayers (disciplines and categories). Each sublayer could be shown\hidden by checkbox. Visibility property contains the state of this checkbox.
At the same time, a sublayer parent’s visibility may be false while the sublayer’s visibility is true. In this case the sublayer should be hidden as well and show property will return false. But the checkbox state (visibility property) will remain true.